From 370157f4932cf9067ba81c8bd5a311aff610882b Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Wed, 02 Dec 2009 20:28:49 +0000
Subject: batch: Avoid flushing a NULL batch

During shutdown from a FatalError during batchbuffer submission, it is
possible for the batch_ptr to be NULL, so we must be careful not to
append a flush on this error path.

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>

Backported to 2.9.1 by Bryce Harrington <bryce@bryceharrington.org>

---
diff -Nurp xserver-xorg-video-intel-2.9.1/src/i830_accel.c xserver-xorg-video-intel-2.9.1-working/src/i830_accel.c
--- xserver-xorg-video-intel-2.9.1/src/i830_accel.c	2009-10-26 04:48:05.000000000 -0700
+++ xserver-xorg-video-intel-2.9.1-working/src/i830_accel.c	2010-03-30 17:24:08.000000000 -0700
@@ -136,7 +136,7 @@ I830Sync(ScrnInfoPtr pScrn)
    if (I810_DEBUG & (DEBUG_VERBOSE_ACCEL | DEBUG_VERBOSE_SYNC))
       ErrorF("I830Sync\n");
 
-   if (!pScrn->vtSema || !pI830->batch_bo)
+   if (!pScrn->vtSema || !pI830->batch_bo || !pI830->batch_ptr)
        return;
 
    I830EmitFlush(pScrn);
